File: Data Structures & Algorithms/eating-bananas/submission-1.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        // First calculate the max value of bananas
        int max = 0;
        for (int i =0; i < piles.Length; i++) {
            max = Math.Max(piles[i], max);
        }

        int low = 1, high = max;
        while(low <= high) {
            int mid = low + (high - low) / 2;
            // Calculate hours required at mid speed
            int totalHours = 0;
            for(int i = 0; i < piles.Length; i++) {
               totalHours += (int)Math.Ceiling((double)piles[i] / mid);
            }
            if(totalHours > h) 
            {
                // This means we are slow and need faster speed
                low = mid + 1;
            }
            else
            {
                // need to find the lowest in the lower half
                high = mid - 1;
            }
        }
        return low;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        map = {}
        for c in s:
            map[c] = map.get(c, 0) + 1

        for c in t:
            if c not in map:
                return False
            else:
                if map[c] == 0:
                    return False
                map[c] = map[c] - 1

        return True